core/errors/desugar.h

```diff
@@ -14,6 +14,7 @@ constexpr ErrorClass UndefUsage{3008, StrictLevel::Strict};
 constexpr ErrorClass UnsupportedRestArgsDestructure{3009, StrictLevel::True};
 constexpr ErrorClass CodeInRBI{3010, StrictLevel::False};
 constexpr ErrorClass DuplicatedHashKeys{3011, StrictLevel::False};
+constexpr ErrorClass BlockAnonymousRestParam{3012, StrictLevel::False};
```

Comment:
You'll notice that we have some more automation around our docs website now.

Adding new error codes requires adding a section documenting the error code in the error-reference.md file now.

(The other automation we've added, which doesn't affect this PR, is that the cli-ref.md document contains the full `--help` output from the current version of Sorbet, which is kept in sync via a Bazel expectation test. https://sorbet.org/docs/cli-ref)

Reply:
That's a nice check, glad it's there!